### PR TITLE
Add pushing image to GCR in github workflow

### PR DESCRIPTION
**Pull request by blackpiglet**
Push to GCR in github workflow to faciliate some environments that have rate limitation to docker hub, e.g. vSphere.

Signed-off-by: Xun Jiang <jxun@vmware.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by sseago — DISMISSED** [review]

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -46,3 +46,6 @@ jobs:
       run: |
         docker login -u ${{ secrets.DOCKER_USER }} -p ${{ secrets.DOCKER_PASSWORD }}
         ./hack/docker-push.sh
+
+        # Push image to GCR to facilitate some environments that have rate limitation to docker hub, e.g. vSphere.
+        REGISTRY=gcr.io/velero-gcp ./hack/docker-push.sh
```
Comment (ywk253100) [inline .github/workflows/push.yml:51]: How do we handle the GCR credential used to push?

Comment (blackpiglet) [reply .github/workflows/push.yml:51]: I missed this part. I did the verification in my already setup environment, so I didn't met the issue.
I will add GCP related action to do the job.

**Comment by blackpiglet** [comment]
Update the procedure to setup Github Action to login to GCP and push images to GCR here temporarily. 

* create service account for GCR image operation: `gcloud iam service-accounts create gcraccount`
* bind GCR administration role to service account: `gcloud projects add-iam-policy-binding velero-gcp --member "serviceAccount:gcraccount@velero-gcp.iam.gserviceaccount.com" --role "roles/storage.objectAdmin"`
* create workload identity pool: `gcloud iam workload-identity-pools create "velero-pool" \
  --project="velero-gcp" \
  --location="global" \
  --display-name="velero pool"`
* create workload identity provider: `gcloud iam workload-identity-pools providers create-oidc "velero-provider" \
  --project="velero-gcp" \
  --location="global" \
  --workload-identity-pool="velero-pool" \
  --display-name="velero provider" \
  --attribute-mapping="google.subject=assertion.sub,attribute.actor=assertion.actor,attribute.aud=assertion.aud" \
  --issuer-uri="https://token.actions.githubusercontent.com"`
* add OIDC Github repository permission to service account: `gcloud iam service-accounts add-iam-policy-binding "gcraccount@velero-gcp.iam.gserviceaccount.com" \
  --project="velero-gcp" \
  --role="roles/iam.workloadIdentityUser" \
  --member="principalSet://iam.googleapis.com/projects/298797809281/locations/global/workloadIdentityPools/velero-pool/attribute.repository/vmware-tanzu/velero"`
* get workload identity pool id: `gcloud iam workload-identity-pools describe 'velero-pool' --project="velero-gcp" --location=global --format
="value(name)"`
* get workload identity provider id: `gcloud iam workload-identity-pools providers describe 'velero-provider' --project="velero-gcp" --location=global --workload-identity-pool="velero-pool" --format="value(name)"`

**Comment by blackpiglet** [comment]
I'm still trying to verify this workflow on my private repo, so it's still in progress. Please leave it here for a moment by now. Sorry for the inconvenience.

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -46,3 +62,28 @@ jobs:
       run: |
         docker login -u ${{ secrets.DOCKER_USER }} -p ${{ secrets.DOCKER_PASSWORD }}
         ./hack/docker-push.sh
+
+    # actions/checkout MUST come before auth
+    - uses: 'actions/checkout@v2'
+
+    # auth to GCP with OIDC token from GCP Workload Identity Provider. Output auth result to access token.
+    - id: 'auth'
```
Comment (reasonerjt) [inline .github/workflows/push.yml:70]: how about use a long lived credential so we do not need to do the token exchange?
https://cloud.google.com/container-registry/docs/advanced-authentication#json-key

Comment (blackpiglet) [reply .github/workflows/push.yml:70]: It should work too. I will check.

**Review by reasonerjt — APPROVED** [review]

**Review by qiuming-best — APPROVED** [review]

**Pull request merged**